txs/conservative_state_test.go

```diff
@@ -693,16 +699,95 @@ func TestApplyLayer_VMError(t *testing.T) {
 	}
 }
 
+func TestConsistentHandling(t *testing.T) {
+	// there are two different workflows for transactions
+	// 1. receive gossiped transaction and verify it immediately
+	// 2. receive synced transaction and delay verification
```

Comment:
i feel we should differentiate synced transaction while gossiping and synced transaction while syncing.
the former case we should still be able to go through the gossip codepath and verify immediately.

Reply:
it is hard for me to tell if thats going to work
the problem is that it is possible to safely verify transaction from layer N, only when layer N-1 was applied

Reply:
i think this will happen naturally when conservative cache is only working after the node is synced.

> the problem is that it is possible to safely verify transaction from layer N, only when layer N-1 was applied

is that true for mempool transactions too? i'd assume that verifying transactions are independent of layer. 
whatever transactions that failed to verify will stay in db (not conservative cache), and when a layer is applied, the conservative cache will query db again for all feasible transactions again.

Reply:
> i think this will happen naturally when conservative cache is only working after the node is synced.

and i meant to work on that next.

Reply:
> is that true for mempool transactions too? i'd assume that verifying transactions are independent of layer.

verification will always fail if account wasn't spawned

> whatever transactions that failed to verify will stay in db (not conservative cache),

so are you going to save all transactions, even those that have failed during gossip? i doubt that it will work

> and i meant to work on that next.

i am pretty sure that you will have to always save transactions that are referenced somewhere without verifying them if layers are applied in the parallel